A Survey on Technology Choice
======


In [2]:
# For nicer printing
options(digits=2);

In [21]:
# Read in the data
data <- read.csv("TechSurvey - Survey.csv",header=T);

#convert date to unix second
for (i in c("Start", "End")) 
    data[,i] = as.numeric(as.POSIXct(strptime(data[,i], "%Y-%m-%d %H:%M:%S")))
for (i in 0:12){
    vnam = paste(c("PG",i,"Submit"), collapse="")
    data[,vnam] = as.numeric(as.POSIXct(strptime(data[,vnam], "%Y-%m-%d %H:%M:%S")))
}
#calculate differences in time    
for (i in 12:0){
    pv = paste(c("PG",i-1,"Submit"), collapse="");
    if (i==0) 
        pv="Start";
    vnam = paste(c("PG",i,"Submit"), collapse="");
    data[,vnam] = data[,vnam] -data[,pv];
}

In [12]:
#now explore variables
summary(data);

     Device    Completed       Start               End               PG0Dis   
        :  2   0    :  2   Min.   :1.54e+09   Min.   :1.54e+09   Min.   :  0  
 Bot    :  1   FALSE:546   1st Qu.:1.54e+09   1st Qu.:1.54e+09   1st Qu.:  0  
 PC     :955   TRUE :805   Median :1.54e+09   Median :1.54e+09   Median :  1  
 Phone  :376               Mean   :1.54e+09   Mean   :1.54e+09   Mean   : 44  
 Tablet : 16               3rd Qu.:1.54e+09   3rd Qu.:1.54e+09   3rd Qu.: 24  
 Unknown:  3               Max.   :1.54e+09   Max.   :1.54e+09   Max.   :168  
                           NA's   :2          NA's   :548        NA's   :73   
    PG0Shown      PG0Submit    
 Min.   :   0   Min.   :    2  
 1st Qu.:   0   1st Qu.:    6  
 Median : 102   Median :    9  
 Mean   : 249   Mean   :  299  
 3rd Qu.: 428   3rd Qu.:   15  
 Max.   :1190   Max.   :76226  
 NA's   :73     NA's   :199    
                                       PG1PsnUse  
 For personal work and/or research use      :727  
          

### Interpret  basic summaries

In [13]:
#get numeric fields only for correlation
sel = c()
for (i in 1:dim(data)[2]) if (is.numeric(data[,i])) sel = c(sel, i);

cor(data[,sel],method="spearman",use="pairwise.complete.obs"); #OK for any: uses ranks

Start   End     PG0Dis  PG0Shown PG0Submit PG1Submit PG2Submit
Start        1.0000  0.9952 -0.0417 -0.11507  0.1350    0.1156    0.0791  
End          0.9952  1.0000 -0.0415 -0.09879  0.1142    0.1550    0.0791  
PG0Dis      -0.0417 -0.0415  1.0000  0.87220  0.0153    0.0065    0.0041  
PG0Shown    -0.1151 -0.0988  0.8722  1.00000  0.0360    0.0205    0.0023  
PG0Submit    0.1350  0.1142  0.0153  0.03596  1.0000    0.1088    0.1037  
PG1Submit    0.1156  0.1550  0.0065  0.02047  0.1088    1.0000    0.1452  
PG2Submit    0.0791  0.0791  0.0041  0.00235  0.1037    0.1452    1.0000  
PG3Submit    0.0384  0.0511  0.0567  0.04968  0.1273    0.2688    0.2045  
PG4Dtr0_6    0.0121 -0.0518  0.1637  0.08226 -0.0080   -0.0685    0.0015  
PG4Psv7_8    0.0037 -0.0458  0.0267  0.00036 -0.0376    0.0566    0.0090  
PG4Prm9_10  -0.0267 -0.0267 -0.0092  0.03279 -0.0939    0.0120   -0.0587  
PG4AllResp   0.0063 -0.0158  0.0018 -0.02094 -0.0236    0.0297    0.0293  
PG4Submit    0.0187  0.0172 -0.0539 -0.05978  0.2191    0.1651    0.1515  
PG5_1Order   0.0218  0.0196  0.0140  0.01254 -0.0240    0.0750   -0.0069  
PG5_2Order   0.0014  0.0002 -0.0386 -0.03617  0.0402   -0.0226   -0.0297  
PG5_3Order  -0.0089 -0.0177  0.0441  0.04228  0.0155    0.0391   -0.0131  
PG5_4Order   0.0931  0.0949 -0.0262 -0.02214 -0.0172    0.0293    0.0560  
PG5_5Order  -0.0523 -0.0466 -0.0087 -0.01058  0.0860    0.0345    0.0570  
PG5_6Order   0.0237  0.0217 -0.0480 -0.04902  0.0762    0.0327    0.1077  
PG5_7Order  -0.0200 -0.0236  0.0220 -0.00444 -0.1174   -0.0815   -0.0285  
PG5_8Order  -0.0804 -0.0852 -0.0147 -0.00433 -0.0421   -0.0205   -0.0387  
PG5_9Order   0.0171  0.0167 -0.0712 -0.10235  0.0476    0.0260   -0.0410  
PG5_10Order -0.0214 -0.0112  0.0200  0.02736  0.0086   -0.0017   -0.0418  
PG5_11Order -0.0241 -0.0196  0.0190  0.02551  0.0927   -0.0047   -0.1043  
PG5_12Order -0.0369 -0.0359  0.0151  0.00743 -0.0161    0.0512    0.0210  
PG5_13Order  0.0598  0.0661  0.0384  0.04072 -0.0280   -0.0651   -0.0047  
PG5Submit    0.0851  0.0909  0.0060 -0.00888  0.1767    0.2467    0.2185  
PG6Submit    0.0054  0.0051  0.0277  0.04005  0.1518    0.2414    0.2696  
PG7Submit    0.0776  0.0759  0.0097  0.01212  0.1365    0.1133    0.1245  
PG8Submit    0.0441  0.0435  0.0354  0.02635  0.1258    0.1069    0.1567  
PG9Submit    0.0410  0.0407  0.0099  0.00056  0.1758    0.1090    0.2013  
PG10Submit   0.0474  0.0517 -0.0293 -0.04481  0.2248    0.1701    0.0989  
PG11Submit   0.0790  0.0792 -0.0454 -0.07102  0.1093    0.0738    0.1147  
PG12Submit   0.0746  0.0772  0.0546  0.04364  0.1096    0.1137    0.1073  
            PG3Submit PG4Dtr0_6 PG4Psv7_8 ... PG5_12Order PG5_13Order PG5Submit
Start        0.0384    0.01210   0.00371  ... -0.0369      0.0598      0.08512 
End          0.0511   -0.05185  -0.04576  ... -0.0359      0.0661      0.09088 
PG0Dis       0.0567    0.16368   0.02668  ...  0.0151      0.0384      0.00601 
PG0Shown     0.0497    0.08226   0.00036  ...  0.0074      0.0407     -0.00888 
PG0Submit    0.1273   -0.00802  -0.03763  ... -0.0161     -0.0280      0.17671 
PG1Submit    0.2688   -0.06852   0.05661  ...  0.0512     -0.0651      0.24670 
PG2Submit    0.2045    0.00146   0.00897  ...  0.0210     -0.0047      0.21851 
PG3Submit    1.0000    0.00865   0.04424  ...  0.0464     -0.0222      0.26048 
PG4Dtr0_6    0.0087    1.00000        NA  ...  0.1774     -0.1289     -0.05214 
PG4Psv7_8    0.0442         NA   1.00000  ... -0.0008     -0.0218      0.08974 
PG4Prm9_10  -0.0062         NA        NA  ...  0.0233      0.0916      0.00077 
PG4AllResp  -0.0193    1.00000   1.00000  ... -0.0306     -0.0166      0.01248 
PG4Submit    0.1956   -0.14272  -0.08350  ... -0.0119     -0.0376      0.27328 
PG5_1Order   0.0578   -0.09488  -0.01399  ... -0.0785     -0.0950      0.08054 
PG5_2Order   0.0048   -0.00954   0.08081  ...  0.0527     -0.0652      0.03898 
PG5_3Order   0.0172    0.13631   0.04396  ... -0.0698     -0.0250      0.23450 
PG5_4Order  -0.0462    0.01

Interpret correlations: onlys start vs End, calculate differene instead


### Simple questions

- Time to take entire survey?
    Time to take the entire survey = PGSubmit time of each quetion added together
                                   = 0.135+.1156+.079.0384+.0817+.0851+.054+0776+.0441+.041+.0474+.0790+.0746
                                   = 0.9115 hours
- Question that took the longest to complete?
    Time to complete = stop - start
    Question 1 time = 0.1550 - 0.1156 = 0.0394
- Question that took the least time?
    Time to complete = stop - start
    Question 11 time = 0.0792 - 0.0790 = 0.0002
- Top-ranked criteria?
    If we are looking at question 5, the selection that was most often considered a high priority was PG5_5PHR which refers to a packages historic reputation. 252 responses said ths was a high priority. 
- Demographic distribution by age?
    18-24: 34
    25-34: 338
    35-44: 258
    45-54: 89
    55-64: 36
    65 and over: 8
    
    The majority of people taking this survey are 25-44. This means that the results of the survey most accurately represents that age group.
    
    Specified Variable: Issue Backlog
    -Hypothesis: That there will be a correlation between age and priority ranking for Issue Backlog
    -According to the survey:
        Essential: 3
        High Priority: 26
        Low Priority: 121
        Medium Priority: 92
        Not a Priority: 188
        NA: 923
    -Based on the results the majority of people taking the survey do not consider issue backlog a priority.
    -Looking into age for each type of response, there is not a noticable grouping of people in an age group for each particular priority level.
    -923 NA responses reduces the ability to find a correlation because there is less data.
    -The lack of an age correlation proves my hypothesis wrong. A specific age group does not have a common response for the question about issue backlog.
    
    -There is not a high correlation between a specific demographic and a particular response to the issue backlog priority.
    
    Since there is not a definite correlation it is hard to choose a specific model that would accurately represent the data.
    
    How could the survey be imporved?
    -Focring the users to give one of the provided responses rather than an NA response would increase the amount of data available for analysis. This could help with the generation of correlations and an accurate model.